astropy/io/fits/hdu/compressed/src/compression.c

```diff
@@ -90,7 +90,10 @@ PyInit__compression(void)
 // cfitsio call.  In our wrapper functions we can then check if the Python error
 // state is set and then return NULL to raise the error.
 void ffpmsg(const char *err_message) {
+    PyGILState_STATE gstate;
+    gstate = PyGILState_Ensure();
     PyErr_SetString(CfitsioException, err_message);
+    PyGILState_Release(gstate);
```

Comment:
Is this assuming that we are always releasing the GIL when calling out to cfitsio? If for some reason that were to change would it explode?

Reply:
Not releasing the GIL would mean no longer being able to do multithreaded decompression of tiles which does work at the moment.

Reply:
So I guess the fix here is fine for now, but we may have to reconsider when subinterpreters become a thing (planned for 3.13) ?

Reply:
> If you need to call Python code from these threads (often this will be part of a callback API provided by the aforementioned third-party library), you must first register these threads with the interpreter by creating a thread state data structure, then acquiring the GIL, and finally storing their thread state pointer, before you can start using the Python/C API. When you are done, you should reset the thread state pointer, release the GIL, and finally free the thread state data structure.

Nope, I am out lol too much C :rofl: